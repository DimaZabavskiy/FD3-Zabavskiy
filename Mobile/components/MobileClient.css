div.MobileClient
{
  padding: 2px 0;
}

.active {
  background-color: green;
  text-align: center;
}
.blocked {
  background-color: red;
  text-align: center;
}