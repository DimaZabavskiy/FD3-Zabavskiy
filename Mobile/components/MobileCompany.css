.MobileCompanyName
{
  font-weight: bold;
}

.MobileCompanyClients
{
  padding-top: 5px;
}
