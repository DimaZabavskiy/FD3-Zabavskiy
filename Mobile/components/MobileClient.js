import React from 'react';
import PropTypes from 'prop-types';
import {voteEvents} from './events';

import './MobileClient.css';

class MobileClient extends React.PureComponent {

  static propTypes = {
    info:PropTypes.shape({
      id: PropTypes.number.isRequired,
      fam: PropTypes.string.isRequired,
      im: PropTypes.string.isRequired,
      otch: PropTypes.string.isRequired,
    }),
  };

  state = {
    info: this.props.info,
  };

  deleteClient = () => {
    voteEvents.emit('deleteClient',this.props.info.id);
  }
  correct = () => {
    voteEvents.emit('correct',this.props.info);
  }

  componentWillReceiveProps = (newProps) => {
    this.setState({info:newProps.info});
  };

  render() {

    console.log("MobileClient id="+this.state.info.id+" render");

    let status;
    if (this.state.info.balance >= 0) {
      status = 'active';
    } else {
      status = 'blocked';
    }
    
    return (      
      <tr>
        <td>{this.state.info.fam}</td> 
        <td>{this.state.info.im}</td> 
        <td>{this.state.info.otch}</td> 
        <td>{this.state.info.balance}</td> 
        <td className={status}>{status}</td> 
        <td>
          <input type="button" value="Редактировать" onClick={this.correct} />
        </td> 
        <td>
          <input type="button" value="Удалить" onClick={this.deleteClient} />
        </td> 
      </tr>
    );

  }

}

export default MobileClient;
