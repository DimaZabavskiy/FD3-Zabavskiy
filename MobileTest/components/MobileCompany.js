import React from 'react';
import PropTypes from 'prop-types';
import {voteEvents} from './events';

import MobileClient from './MobileClient';

//import './MobileCompany.css';

class MobileCompany extends React.PureComponent {

  static propTypes = {
    clients:PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.number.isRequired,
        fam: PropTypes.string.isRequired,
        im: PropTypes.string.isRequired,
      })
    ),
  };

  state = {
    clients: this.props.clients,
    clientStatus: 'common',
    workMode: 1,
    correctClient: null,
    correctActive: false,
    startNewClient: false,
    newClientFam: null,
    newClientIm: null,
    newClientOtch: null,
    newClientBalance: null,
  };

  componentDidMount = () => {
    voteEvents.addListener('deleteClient',this.deleteClient);
    voteEvents.addListener('correct',this.correct);
  };
  componentWillUnmount = () => {
    voteEvents.removeListener('deleteClient',this.deleteClient);
    voteEvents.removeListener('correct',this.correct);
  };

  deleteClient = (code) => {
    if (!this.state.startNewClient && !this.state.correctActive) {
      let newClients=[]; // копия самого массива клиентов
      this.state.clients.forEach( (c,i) => {
        if ( c.id !=code ) {
          newClients.push(c);
        }
      } );
      this.setState({clients:newClients});
    }
  }

  showAll = () => {
    this.setState({
      clientStatus: 'common'})
  }

  showActive = () => {
    this.setState({
      clientStatus: 'active'})
  }

  showBlocked = () => {
    this.setState({
      clientStatus: 'blocked'})
  }

  addClient = () => {
    if (this.state.correctClient) {
    return
  }
    this.setState({workMode: 2, startNewClient: true})
  }

  correct = (client) => {
    if (this.state.startNewClient || this.state.correctActive) {
      return
    }
    this.setState({workMode:2, correctClient:client, correctActive: true});
  }

  addOrCorrectClient = () => {
    let newClients=[...this.state.clients];
    if (!this.state.correctActive) {
      let id = Math.random();
      newClients.push({id:id, fam:this.state.newClientFam, im:this.state.newClientIm,
        otch:this.state.newClientOtch, balance:this.state.newClientBalance}, )
      this.setState({workMode: 1,clients:newClients, correctClient:null, correctActive:false,
        newClientFam: null, newClientIm: null, newClientOtch: null, newClientBalance: null, startNewClient: false})
    } else {
      for (let i = 0; i < newClients.length; i++) {
        if (newClients[i].id == this.state.correctClient.id) {
          newClients[i] = this.state.correctClient;
          this.setState({workMode: 1, clients:newClients, correctClient:null, correctActive:false,
          newClientFam: null, newClientIm: null, newClientOtch: null, newClientBalance: null,startNewClient: false})
        }
      }
    }    
  }

  cancelClient = () => {
    this.setState({workMode: 1, correctClient:null, correctActive: false, 
      newClientFam: null, newClientIm: null, newClientOtch: null, newClientBalance: null,startNewClient: false})
  }

  changedFam = (EO) => {
    if(this.state.correctActive) {
      let correctClient = {...this.state.correctClient};
      correctClient.fam = EO.target.value;
      this.setState({correctClient:correctClient});
    } else {
      this.setState({newClientFam:EO.target.value});
    }
  }
  changedIm = (EO) => {
    if(this.state.correctActive) {
      let correctClient = {...this.state.correctClient};
      correctClient.im = EO.target.value;
      this.setState({correctClient:correctClient});
    } else {
      this.setState({newClientIm:EO.target.value});
    }
  }
  changedOtch = (EO) => {
    if(this.state.correctActive) {
      let correctClient = {...this.state.correctClient};
      correctClient.otch = EO.target.value;
      this.setState({correctClient:correctClient});
    } else {
      this.setState({newClientOtch:EO.target.value});
    }
  }
  changedBalance = (EO) => {
    if(this.state.correctActive) {
      let correctClient = {...this.state.correctClient};
      correctClient.balance = EO.target.value;
      this.setState({correctClient:correctClient});
    } else {
      this.setState({newClientBalance:EO.target.value});
    }
  }
  
  render() {

    //console.log("MobileCompany render");

    var clientsCode=[];

    this.state.clients.forEach( client => {
      if (this.state.clientStatus == 'common') {
        clientsCode.push(<MobileClient key={client.id} info={client}  />)
      } else if (this.state.clientStatus == 'active') {
        if (client.balance >=0) {
          clientsCode.push(<MobileClient key={client.id} info={client}  />)
        }
      } else if (this.state.clientStatus == 'blocked') {
        if (client.balance < 0) {
          clientsCode.push(<MobileClient key={client.id} info={client}  />)
        }
      }
    });

    return (
      <div className='MobileCompany'>
        <input type="button" value="Все" onClick={this.showAll} />
        <input type="button" value="Активные" onClick={this.showActive} />
        <input type="button" value="Заблокированные" onClick={this.showBlocked} />

        <hr />

        <table>
          <thead>
            <tr>
              <th>Фамилия</th>
              <th>Имя</th>
              <th>Отчество</th>
              <th>Баланс</th>
              <th>Статус</th>
              <th>Редактировать</th>
              <th>Удалить</th>
            </tr>
          </thead>

          <tbody>{clientsCode}</tbody>
          
        </table>
        
        <hr />

        <input type="button" value="Добавить клиента" onClick={this.addClient} />

        {(this.state.workMode == 2) &&
        <div>
          Фамилия: <input type='text' value={(this.state.correctClient)? this.state.correctClient.fam:
          ((this.state.newClientFam)? this.state.newClientFam: ' ')}
          onChange={this.changedFam} />
          <br />
          Имя: <input type='text' value={(this.state.correctClient)? this.state.correctClient.im:
          ((this.state.newClientIm)? this.state.newClientIm: ' ')}
          onChange={this.changedIm} />
          <br />
          Отчество: <input type='text' value={(this.state.correctClient)? this.state.correctClient.otch:
          ((this.state.newClientOtch)? this.state.newClientOtch: ' ')}
          onChange={this.changedOtch} />
          <br />
          Баланс: <input type='text' value={(this.state.correctClient)? this.state.correctClient.balance:
          ((this.state.newClientBalance)? this.state.newClientBalance: ' ')}
          onChange={this.changedBalance} />
          <br />
          <input type='button' value='Добавить' onClick={this.addOrCorrectClient} />
          <input type='button' value='Отмена' onClick={this.cancelClient} />
        </div>
        }

      </div>
    )
    ;

  }

}

export default MobileCompany;
