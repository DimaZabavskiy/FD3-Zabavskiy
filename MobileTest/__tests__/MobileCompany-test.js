"use strict";

import React from 'react';
import renderer from 'react-test-renderer';

import MobileCompany from '../components/MobileCompany';
import MobileClient from '../components/MobileClient';

test('работа MobileCompany', () => {
  let clientsArr=[ 
    {id:101, fam:"Иванов", im:"Иван", otch:"Иванович", balance:200}, 
    {id:105, fam:"Сидоров", im:"Сидор", otch:"Сидорович", balance:-250}, 
    {id:110, fam:"Петров", im:"Пётр", otch:"Петрович", balance:180},
    {id:120, fam:"Григорьев", im:"Григорий", otch:"Григорьевич", balance:-220},
  ];

  // создаём тестовую версию компонента
  const component = renderer.create(
    <MobileCompany clients={clientsArr}/>
  );

  // получаем снэпшот (HTML-снимок) компонента для сверки, что вёрстка не испортилась
  let componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();

  // найдём в вёрстке компонента кнопку 'Все'
  const buttonElem = component.root.find( el => el.type=='input' && el.props.value == 'Все' ); 
  buttonElem.props.onClick();

  // получаем уже изменённый снэпшот
  componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();

  
  // найдём в вёрстке компонента кнопку 'Активные'
  const buttonElemActive = component.root.find( el => el.type=='input' && el.props.value == 'Активные' ); 
  buttonElemActive.props.onClick();

  componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();

  // найдём в вёрстке компонента кнопку 'Заблокированные'
  const buttonElemBlocked = component.root.find( el => el.type=='input' && el.props.value == 'Заблокированные' ); 
  buttonElemBlocked.props.onClick();

  componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();

  const component2 = renderer.create(
    <MobileClient key={clientsArr[1].id} info={clientsArr[1]}/>
  );

  // найдём в вёрстке компонента кнопку 'Удалить'
  const buttonDelete = component2.root.find( el => el.type=='input' && el.props.value == 'Удалить' );
  buttonDelete.props.onClick();

  componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();

  const component3 = renderer.create(
    <MobileClient key={clientsArr[3].id} info={clientsArr[3]}/>
  );

  // найдём в вёрстке компонента кнопку 'Редактировать'
  const buttonCorrect = component3.root.find( el => el.type=='input' && el.props.value == 'Редактировать' );
  buttonCorrect.props.onClick();

  componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();

  // найдём в вёрстке компонента кнопку 'Добавить изменения'
  const buttonElemChange = component.root.find( el => el.type=='input' && el.props.value == 'Добавить' ); 
  buttonElemChange.props.onClick();

  componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();

  // найдём в вёрстке компонента кнопку 'Добавить клиента'
  const buttonElemAdd = component.root.find( el => el.type=='input' && el.props.value == 'Добавить клиента' ); 
  buttonElemAdd.props.onClick();
  
  componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();

  // найдём в вёрстке компонента кнопку 'Отмена'
  const buttonElemCancel = component.root.find( el => el.type=='input' && el.props.value == 'Отмена' ); 
  buttonElemCancel.props.onClick();

  componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();
  
});
