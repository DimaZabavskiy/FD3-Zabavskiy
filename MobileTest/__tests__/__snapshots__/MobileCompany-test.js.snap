// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`работа MobileCompany 1`] = `
<div
  className="MobileCompany"
>
  <input
    onClick={[Function]}
    type="button"
    value="Все"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Активные"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Заблокированные"
  />
  <hr />
  <table>
    <thead>
      <tr>
        <th>
          Фамилия
        </th>
        <th>
          Имя
        </th>
        <th>
          Отчество
        </th>
        <th>
          Баланс
        </th>
        <th>
          Статус
        </th>
        <th>
          Редактировать
        </th>
        <th>
          Удалить
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Иванов
        </td>
        <td>
          Иван
        </td>
        <td>
          Иванович
        </td>
        <td>
          200
        </td>
        <td
          className="active"
        >
          active
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr>
        <td>
          Сидоров
        </td>
        <td>
          Сидор
        </td>
        <td>
          Сидорович
        </td>
        <td>
          -250
        </td>
        <td
          className="blocked"
        >
          blocked
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr>
        <td>
          Петров
        </td>
        <td>
          Пётр
        </td>
        <td>
          Петрович
        </td>
        <td>
          180
        </td>
        <td
          className="active"
        >
          active
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr>
        <td>
          Григорьев
        </td>
        <td>
          Григорий
        </td>
        <td>
          Григорьевич
        </td>
        <td>
          -220
        </td>
        <td
          className="blocked"
        >
          blocked
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <hr />
  <input
    onClick={[Function]}
    type="button"
    value="Добавить клиента"
  />
</div>
`;

exports[`работа MobileCompany 2`] = `
<div
  className="MobileCompany"
>
  <input
    onClick={[Function]}
    type="button"
    value="Все"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Активные"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Заблокированные"
  />
  <hr />
  <table>
    <thead>
      <tr>
        <th>
          Фамилия
        </th>
        <th>
          Имя
        </th>
        <th>
          Отчество
        </th>
        <th>
          Баланс
        </th>
        <th>
          Статус
        </th>
        <th>
          Редактировать
        </th>
        <th>
          Удалить
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Иванов
        </td>
        <td>
          Иван
        </td>
        <td>
          Иванович
        </td>
        <td>
          200
        </td>
        <td
          className="active"
        >
          active
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr>
        <td>
          Сидоров
        </td>
        <td>
          Сидор
        </td>
        <td>
          Сидорович
        </td>
        <td>
          -250
        </td>
        <td
          className="blocked"
        >
          blocked
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr>
        <td>
          Петров
        </td>
        <td>
          Пётр
        </td>
        <td>
          Петрович
        </td>
        <td>
          180
        </td>
        <td
          className="active"
        >
          active
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr>
        <td>
          Григорьев
        </td>
        <td>
          Григорий
        </td>
        <td>
          Григорьевич
        </td>
        <td>
          -220
        </td>
        <td
          className="blocked"
        >
          blocked
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <hr />
  <input
    onClick={[Function]}
    type="button"
    value="Добавить клиента"
  />
</div>
`;

exports[`работа MobileCompany 3`] = `
<div
  className="MobileCompany"
>
  <input
    onClick={[Function]}
    type="button"
    value="Все"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Активные"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Заблокированные"
  />
  <hr />
  <table>
    <thead>
      <tr>
        <th>
          Фамилия
        </th>
        <th>
          Имя
        </th>
        <th>
          Отчество
        </th>
        <th>
          Баланс
        </th>
        <th>
          Статус
        </th>
        <th>
          Редактировать
        </th>
        <th>
          Удалить
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Иванов
        </td>
        <td>
          Иван
        </td>
        <td>
          Иванович
        </td>
        <td>
          200
        </td>
        <td
          className="active"
        >
          active
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr>
        <td>
          Петров
        </td>
        <td>
          Пётр
        </td>
        <td>
          Петрович
        </td>
        <td>
          180
        </td>
        <td
          className="active"
        >
          active
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <hr />
  <input
    onClick={[Function]}
    type="button"
    value="Добавить клиента"
  />
</div>
`;

exports[`работа MobileCompany 4`] = `
<div
  className="MobileCompany"
>
  <input
    onClick={[Function]}
    type="button"
    value="Все"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Активные"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Заблокированные"
  />
  <hr />
  <table>
    <thead>
      <tr>
        <th>
          Фамилия
        </th>
        <th>
          Имя
        </th>
        <th>
          Отчество
        </th>
        <th>
          Баланс
        </th>
        <th>
          Статус
        </th>
        <th>
          Редактировать
        </th>
        <th>
          Удалить
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Сидоров
        </td>
        <td>
          Сидор
        </td>
        <td>
          Сидорович
        </td>
        <td>
          -250
        </td>
        <td
          className="blocked"
        >
          blocked
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr>
        <td>
          Григорьев
        </td>
        <td>
          Григорий
        </td>
        <td>
          Григорьевич
        </td>
        <td>
          -220
        </td>
        <td
          className="blocked"
        >
          blocked
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <hr />
  <input
    onClick={[Function]}
    type="button"
    value="Добавить клиента"
  />
</div>
`;

exports[`работа MobileCompany 5`] = `
<div
  className="MobileCompany"
>
  <input
    onClick={[Function]}
    type="button"
    value="Все"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Активные"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Заблокированные"
  />
  <hr />
  <table>
    <thead>
      <tr>
        <th>
          Фамилия
        </th>
        <th>
          Имя
        </th>
        <th>
          Отчество
        </th>
        <th>
          Баланс
        </th>
        <th>
          Статус
        </th>
        <th>
          Редактировать
        </th>
        <th>
          Удалить
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Григорьев
        </td>
        <td>
          Григорий
        </td>
        <td>
          Григорьевич
        </td>
        <td>
          -220
        </td>
        <td
          className="blocked"
        >
          blocked
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <hr />
  <input
    onClick={[Function]}
    type="button"
    value="Добавить клиента"
  />
</div>
`;

exports[`работа MobileCompany 6`] = `
<div
  className="MobileCompany"
>
  <input
    onClick={[Function]}
    type="button"
    value="Все"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Активные"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Заблокированные"
  />
  <hr />
  <table>
    <thead>
      <tr>
        <th>
          Фамилия
        </th>
        <th>
          Имя
        </th>
        <th>
          Отчество
        </th>
        <th>
          Баланс
        </th>
        <th>
          Статус
        </th>
        <th>
          Редактировать
        </th>
        <th>
          Удалить
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Григорьев
        </td>
        <td>
          Григорий
        </td>
        <td>
          Григорьевич
        </td>
        <td>
          -220
        </td>
        <td
          className="blocked"
        >
          blocked
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <hr />
  <input
    onClick={[Function]}
    type="button"
    value="Добавить клиента"
  />
  <div>
    Фамилия: 
    <input
      onChange={[Function]}
      type="text"
      value="Григорьев"
    />
    <br />
    Имя: 
    <input
      onChange={[Function]}
      type="text"
      value="Григорий"
    />
    <br />
    Отчество: 
    <input
      onChange={[Function]}
      type="text"
      value="Григорьевич"
    />
    <br />
    Баланс: 
    <input
      onChange={[Function]}
      type="text"
      value={-220}
    />
    <br />
    <input
      onClick={[Function]}
      type="button"
      value="Добавить"
    />
    <input
      onClick={[Function]}
      type="button"
      value="Отмена"
    />
  </div>
</div>
`;

exports[`работа MobileCompany 7`] = `
<div
  className="MobileCompany"
>
  <input
    onClick={[Function]}
    type="button"
    value="Все"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Активные"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Заблокированные"
  />
  <hr />
  <table>
    <thead>
      <tr>
        <th>
          Фамилия
        </th>
        <th>
          Имя
        </th>
        <th>
          Отчество
        </th>
        <th>
          Баланс
        </th>
        <th>
          Статус
        </th>
        <th>
          Редактировать
        </th>
        <th>
          Удалить
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Григорьев
        </td>
        <td>
          Григорий
        </td>
        <td>
          Григорьевич
        </td>
        <td>
          -220
        </td>
        <td
          className="blocked"
        >
          blocked
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <hr />
  <input
    onClick={[Function]}
    type="button"
    value="Добавить клиента"
  />
</div>
`;
